sdk/identity/identity/migration-v1-v2.md

```diff
@@ -0,0 +1,309 @@
+
+## Dependency changes
+
+Version 2 of Azure Identity for JavaScript no longer includes native dependencies (neither ordinary, peer, nor optional dependencies). Previous distributions of `@azure/identity` included an optional dependency on `keytar`, which caused issues for some users in restrictive environments.
+
+Version 2 of Azure Identity for JavaScript now also depends on the latest available versions of `@azure/msal-common`, `@azure/msal-node`, and `@azure/msal-browser`. Our goal is to always be up-to-date with the MSAL versions.
+
+All credentials provided in version 2 of the Identity package for Node.js now make use of the latest `@azure/msal-node`. `InteractiveBrowserCredential` now uses the latest `@azure/msal-node` on Node.js and the latest `@azure/msal-browser` on the browser.
+
+Relying on the latest versions of MSAL comes with many benefits, but it has some considerations:
+
+- For every new instance of all of the updated credentials, now they will do a first request to discover relevant endpoint metadata information from Azure.
+- On the browser, the `InteractiveBrowserCredential` will now use the [Auth Code Flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow) with [PKCE](https://tools.ietf.org/html/rfc7636) rather than [Implicit Grant Flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-implicit-grant-flow) to better support browsers with enhanced security restrictions. Read more on this in our [docs on Interactive Browser Credential](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/interactive-browser-credential.md).
+
+## Bug fixes
+
+The v2 release fixes the following issues:
+
+- `InteractiveBrowserCredential` on Node.js sometimes caused the process to hang if there was no browser available.
+- The `AZURE_AUTHORITY_HOST` environment variable wasn't properly picked up in Node.js.
+
+Furthermore, `ClientSecretCredential`, `ClientCertificateCredential`, and `UsernamePasswordCredential` now throw if the required parameters aren't provided (even in JavaScript).
+
+## Errors
+
+For our version two of Identity, we've renamed the error `CredentialUnavailable` to `CredentialUnavailableError`, to align with the naming convention used for error classes in the Azure SDKs in JavaScript.
+
+We've also introduced a new error, named `AuthenticationRequiredError`. This error will show up when a credential fails to authenticate silently and has the same impact on `ChainedTokenCredential` as the `CredentialUnavailableError`, which is to allow the next credential in the chain to be tried.
+
+Most importantly, errors and logged exceptions may now point to the new [troubleshooting guidelines](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/Troubleshooting.md).
+
+## New credentials
+
+Version 2 of the `@azure/identity` package includes three new credential types:
```

Comment:
We will have more credentials in the future in v2. Unless we keep updating this doc every time we add a new credential, this section will get stale.

```suggestion
As of October 2021, version 2 of the `@azure/identity` package includes the below new credential types. More will be added in the future.
```